src/nb_internals.cpp

```diff
@@ -72,6 +73,12 @@
 #  define NB_STABLE_ABI ""
 #endif
 
+#if defined(NB_FREE_THREADED)
```

Comment:
What happens if someone defines NB_FREE_THREADED but the interpreter doesn't support free-threading? I guess it probably doesn't build due to the use of FT-specific APIs, but can we get a clean #error somewhere about it? Seems like a potential failure mode especially for users of other build systems.

Reply:
Is it worth even having a separate macro from `Py_GIL_DISABLED` - maybe that can replace `NB_FREE_THREADED`?

Reply:
> Seems like a potential failure mode especially for users of other build systems.

I will add an error message.

> Is it worth even having a separate macro from `Py_GIL_DISABLED` - maybe that can replace `NB_FREE_THREADED`?

I wanted these two to be separate so that users can build non-free-threaded extensions (`NB_FREE_THREADED` unset) for a free-threaded interpreter (`Py_GIL_DISABLED` set). Not very useful, but an intermediate step towards porting.